tools/mason/MasonExternal.chpl

```diff
@@ -18,8 +18,8 @@
  * limitations under the License.
  */
 
-/* Version as of Chapel 1.22 - to be updated each release */
-const spackVersion = '0.15.0';
+/* Version as of Chapel 1.25 - to be updated each release */
+const spackVersion = '0.16.2';
```

Comment:
I think we should check to make sure the patch level is respected, because right now it looks like we only check out the Spack release branch based on major and minor. Either that or the patch level is just not important.

Reply:
I added a helper function to compare spack versions down to the patch version, along with tests for that function.
There is now a new path in the control flow for a version of spack that is newer than the one we specified, warning the user that they may experience unexpected behavior.
I have also added a check after the setup to make sure the spack version that we find is the spack version we intended.